src/gazebo_plugins/ControlBoard.cc

```diff
@@ -109,41 +116,62 @@ GZ_REGISTER_MODEL_PLUGIN(GazeboYarpControlBoard)
 
         for (int n = 0; n < netList->size(); n++)
         {
-            yarp::os::ConstString driverName( netList->get(n).asString().c_str());
+            m_driverName = netList->get(n).asString();
```

Comment:
It make sense. An easy trick to avoid complicating code is creating a unique "scoped" device name as `deviceScopedName = robotScopedName+"::" + deviceName` .